### ISSUE TITLE
Add support for RandomGenerator in Java 17

### ISSUE THREAD

**Issue by fabiolimace** [f4b6a3/uuid-creator#70]

**Comment by fabiolimace** [comment]
Now it is easy to use `RandomGenerator` (JDK17) and `SplittableRandom` (JDK8).

Example:

```
// use a random function that returns a long value
RandomGenerator random = RandomGenerator.getDefault();
RandomBasedFactory factory = new RandomBasedFactory(() -> random.nextLong());
```

**Comment by fabiolimace** [comment]
Released v5.0.0.
